src/editors/containers/VideoUploadEditor/index.test.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import { render, fireEvent, act } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import VideoUploadEditor, { VideoUploader } from '.';
+
+const mockDispatch = jest.fn();
+const mockOnUpload = jest.fn();
+
+jest.mock('react-redux', () => ({
+  useDispatch: () => mockDispatch,
+}));
+
+const defaultEditorProps = {
+  intl: {},
+};
+
+const defaultUploaderProps = {
+  onUpload: mockOnUpload,
+  errorMessage: '',
+  intl: {},
+};
+
+const renderEditorComponent = (props = defaultEditorProps) => render(
+  <IntlProvider locale="en">
+    <VideoUploadEditor {...props} />
+  </IntlProvider>,
+);
+
+const renderUploaderComponent = (props = defaultUploaderProps) => render(
+  <IntlProvider locale="en">
+    <VideoUploader {...props} />
+  </IntlProvider>,
+);
+
+describe('VideoUploadEditor', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders without errors', () => {
+    const { container } = renderEditorComponent();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders the input field for pasting the video ID or URL', () => {
+    const { getByPlaceholderText } = renderEditorComponent();
+    const input = getByPlaceholderText('Paste your video ID or URL');
+    expect(input).toBeInTheDocument();
+  });
+
+  it('updates the input field value when user types', () => {
+    const { getByPlaceholderText } = renderEditorComponent();
+    const input = getByPlaceholderText('Paste your video ID or URL');
+
+    fireEvent.change(input, { target: { value: 'test value' } });
+    expect(input.value).toBe('test value');
+  });
+
+  it('shows error message with unsupported files', async () => {
+    const { getByTestId, findByText } = renderEditorComponent();
+    const fileInput = getByTestId('fileInput');
+
+    const unsupportedFile = new File(['(⌐□_□)'], 'unsupported.avi', { type: 'video/avi' });
+    fireEvent.change(fileInput, { target: { files: [unsupportedFile] } });
+
+    const errorMsg = await findByText('Video must be an MP4 or MOV file');
+    expect(errorMsg).toBeInTheDocument();
+  });
```

Comment:
@pkulkark Can you add a similar test, but with a supported file?

Reply:
Sure, will add it.

Reply:
Done.